packages/starlight/__tests__/remark-rehype/rehype-tabs.test.ts

```diff
@@ -1,14 +1,22 @@
 import { expect, test } from 'vitest';
+import { fromHtml } from 'hast-util-from-html';
+import { selectAll } from 'hast-util-select';
+import { rehype } from 'rehype';
 import { processPanels, TabItemTagname } from '../../user-components/rehype-tabs';
 
+const processor = rehype().data('settings', { fragment: true });
+
 const TabItem = ({ label, slot, icon }: { label: string; slot: string; icon?: string }) => {
 	const iconAttr = icon ? ` data-icon="${icon}"` : '';
 	return `<${TabItemTagname} data-label="${label}"${iconAttr}>${slot}</${TabItemTagname}>`;
 };
 
-/** Get an array of HTML strings, one for each `<section>` created by rehype-tabs for each tab item. */
-const extractSections = (html: string) =>
-	[...html.matchAll(/<section.*?<\/section>/g)].map(([section]) => section);
+/** Get an array of HTML strings, one for each `<div role="tabpanel">` created by rehype-tabs for each tab item. */
+const extractTabPanels = (html: string) => {
+	const tree = fromHtml(html, { fragment: true });
+	const tabPanels = selectAll('div[role="tabpanel"]', tree);
+	return tabPanels.map((tabPanel) => processor.stringify({ type: 'root', children: [tabPanel] }));
+};
```

Comment:
Considering we're now using a `<div>` element and the tests contain nested `<div>` elements, I didn't want to rely and maintain a regular expression to extract the panel HTML 😅